From bc85c6410f9154895d47805fe9c5e65402581dd8 Mon Sep 17 00:00:00 2001
From: Evan Wang <xswang@marvell.com>
Date: Tue, 8 Aug 2017 16:08:27 +0800
Subject: [PATCH 1158/1345] dts: mcbin: pcie: add phandle of pcie reset gpio

commit  52557117532c7b63c6f8cccbd618ad66c8e84689 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

MCbin board needs PCIe reset, this patch adds phandle
to reset the PCIe.

Change-Id: Iec455931db766cb8a8b541ab7fedcbd1c6bc7395
Signed-off-by: Evan Wang <xswang@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/42744
Reviewed-by: Igal Liberman <igall@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../dts/marvell/armada-8040-mcbin-pp2-vanilla.dts  |    7 +++++++
 arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts  |    3 +++
 2 files changed, 10 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin-pp2-vanilla.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin-pp2-vanilla.dts
index 56aa640..820e5b6 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin-pp2-vanilla.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin-pp2-vanilla.dts
@@ -95,6 +95,10 @@
 					marvell,pins = "mpp47";
 					marvell,function = "gpio";
 				};
+				pcie_reset_pins: pcie-reset-pins {
+					marvell,pins = "mpp52";
+					marvell,function = "gpio";
+				};
 			};
 			comphy0: comphy {
 				status = "disabled";
@@ -211,6 +215,9 @@
 				&comphy0 3 COMPHY_PCIE0
 				>;
 			phy-names = "pcie-phy0", "pcie-phy1", "pcie-phy2", "pcie-phy3";
+			reset-gpios = <&cpm_gpio1 20 GPIO_ACTIVE_HIGH>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&pcie_reset_pins>;
 		};
 		reg_usb3h0_vbus: usb3-vbus0 {
 			compatible = "regulator-fixed";
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
index 4cc60f3..05c6491 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
@@ -132,6 +132,9 @@
 				&comphy0 3 COMPHY_PCIE0
 				>;
 			phy-names = "pcie-phy0", "pcie-phy1", "pcie-phy2", "pcie-phy3";
+			reset-gpios = <&cpm_gpio1 20 GPIO_ACTIVE_HIGH>;
+			pinctrl-names = "default";
+			pinctrl-0 = <&pcie_reset_pins>;
 		};
 		vccq_sdio_reg: regulator@1 {
 			compatible = "regulator-fixed";
-- 
1.7.9.5

